src/reflect/scala/reflect/internal/tpe/GlbLubs.scala

```diff
@@ -184,8 +184,8 @@ private[internal] trait GlbLubs {
   /** Eliminate from list of types all elements which are a supertype
     *  of some other element of the list. */
   private def elimSuper(ts: List[Type]): List[Type] = ts match {
```

Comment:
`Type`-s aren't amenable to use in a `HashSet` because `=:=` isn't compatible with `hashCode`.  This code here is actually using subtyping, which makes it harder again to find ways to beat `O(N^2)`.